source/common/network/io_socket_handle_impl.h

```diff
@@ -64,8 +64,16 @@ class IoSocketHandleImpl : public IoHandle, protected Logger::Loggable<Logger::I
   absl::optional<int> domain() override;
   Address::InstanceConstSharedPtr localAddress() override;
   Address::InstanceConstSharedPtr peerAddress() override;
-  Event::FileEventPtr createFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
-                                      Event::FileTriggerType trigger, uint32_t events) override;
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
```

Comment:
Did you consider returning a different type of RAII handle here? This would avoid having to remember to call `resetFileEvents()` in various cases. I'm not sure if this is worth the extra memory allocation though.

Reply:
What type of RAII handle do you have in mind?

Reply:
Basically just an "initialized file event" handle which on destruction does the equivalent of the reset function below. I'm not sure this is any better though since I think in all of the cases where you need to reset you need to do it explicitly either way. I was mostly just trying to brainstorm if there are any ways to make this a bit less fragile in terms of cleanup.

Reply:
I have put some thought into this because it also annoyed me that as a consumer of the API I need to remember to call `resetFileEvents()`. But I realized that in most cases (in the previous implementation) the listeners manually reset the pointer.

It might make sense to add this into a destructor as a fail-safe mechanism for when the object gets destroyed.

Reply:
Yeah that's the same conclusion I came to. I don't think it buys you very much so this seems better.